### PR TITLE
Remove extendr-engine call from non-testing code.

### LINKED ISSUE #144: Some tests are not conditionally compiled
Currently the integration tests trigger a note that the compiled code contains non-API entry points:
https://github.com/extendr/extendr/pull/141/checks?check_run_id=1755212957#step:14:90

This isn't reported as an error, so tests seem to pass, but it would actually prevent us from submitting a package to CRAN, so it needs to be fixed.

One case where we have tests without conditional compilation is here:
https://github.com/extendr/extendr/blob/f43d2539e8ac0d5dcd562aa1d2282369528ae1a2/extendr-api/src/robj_ndarray.rs#L59-L61

There may be others. These can be difficult to find. It would be good to have some sort of automated test for these, so they don't sneak in to the code base unnoticed.

### PR DESCRIPTION
**Pull request by clauswilke**
Fixes #144.

### CODE REVIEW

File: extendr-api/src/robj_ndarray.rs [hunk 1]
```diff
@@ -56,6 +56,7 @@ make_array_view_2!(f64, "Not a floating point matrix.");
 //     }
 // }
 
+#[cfg(test)]
```
Comment (yutannihilation) [inline extendr-api/src/robj_ndarray.rs:59]: I don't think this line is needed.

**Review by yutannihilation — APPROVED** [review]
Wow, so simple! Thanks for taking this.

File: extendr-api/src/robj_ndarray.rs (hunk 1 above)
Comment (clauswilke) [reply extendr-api/src/robj_ndarray.rs:59]: Wouldn't an R package that turns on the ndarray feature run into the same problem? We don't currently do that in extendrtests, but we probably should.

Comment (yutannihilation) [reply extendr-api/src/robj_ndarray.rs:59]: I meant, as there's a bare function here and it's marked `#[test]`, the compilation happens only on testing. When we start using test module, we'll need `#[cfg(test)]`.

Let's confirm it here: https://github.com/yutannihilation/extendr/commit/189edba82acdd4906704112981523ef45dc7ec6e

Comment (clauswilke) [reply extendr-api/src/robj_ndarray.rs:59]: Ah, Ok, yes that seems to be correct.

**Pull request merged**